dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -92,6 +123,103 @@ public Entry updateAliases(@ApiParam(hidden = true) @Auth User user,
         return this.toolDAO.findCollectionsByEntryId(entry.getId());
     }
 
+    @PUT
+    @Path("/{id}/topic")
+    @Timed
+    @RolesAllowed({ "curator" })
+    @UnitOfWork
+    @ApiOperation(value = "Create a discourse topic for an entry.", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Entry.class)
+    public Entry setDiscourseTopic(@ApiParam(hidden = true) @Auth User user,
+            @ApiParam(value = "The id of the entry to add a topic to.", required = true) @PathParam("id") Long id,
+            @ApiParam(value = "The id of the category to add a topic to, defaults to Automatic Tool and Workflow Threads (6).", defaultValue = "6", allowableValues = "6,9") @QueryParam("categoryId") Integer categoryId,
```

Comment:
if possible, try using your constant in here somehow

Reply:
No luck getting this working.